addons/website_slides/models/slide_channel.py

```diff
@@ -695,68 +811,123 @@ def action_redirect_to_members(self, completed=False):
             default_email_layout_xmlid='website_slides.mail_notification_channel_invite',
+            default_with_enrollment=with_enrollment,
         )
+        if with_enrollment:
+            name = _('Enroll Attendees to "%(course_name)s"', course_name=course_name) if course_name else _('Enroll Attendees to a course')
+        else:
+            name = _('Invite Attendees to "%(course_name)s"', course_name=course_name) if course_name else _('Invite Attendees to a course')
+
         return {
             'type': 'ir.actions.act_window',
-            'view_mode': 'form',
+            'views': [[False, 'form']],
             'res_model': 'slide.channel.invite',
             'target': 'new',
             'context': local_context,
+            'name': name,
         }
 
     def _action_add_member(self):
-        """ Adds the logged in user in the channel members.
-        (see '_action_add_members' for more info)
+        """ Enroll the logged in user in the course. (see '_action_add_members' for more info)
+        This method is called when a user press the 'Join the course' button in front-end,
+        through '/enroll' route.
+
+        If the logged in user has been invited, then we sudo the add_members method
+        since we consider he has been granted access to the course.
+
+        Returns True if added and enrolled successfully, False otherwise."""
+        invited_res = False
+        courses_invited = self.filtered(lambda course: course.is_member_invited)
+        if courses_invited:
+            invited_res = bool(courses_invited.sudo()._action_add_members(self.env.user.partner_id))
+
+        return invited_res or bool((self - courses_invited)._action_add_members(self.env.user.partner_id))
+
+    def _action_add_members(self, target_partners, member_status='joined'):
+        """ Adds the target_partners as attendees of the channel(s).
+            Partners are added as follows, depending on the value of member_status:
+            1) (Default) 'joined'. The partners will be added as enrolled attendees. This will make the content
+                (slides) of the channel available to that partner. This can also happen when an invited attendee
+                enrolls themself. The attendes are also subscribed to the chatter of the channel.
+            2) 'invited' : This is used when inviting partners. The partners are added as invited attendees
+                This will make the channel accessible but not the slides until they enroll themselves.
+
+        Returns the union of new partners and of invited partners having enrolled themselves."""
+        # Channels on which the current user has the rights to write.
+        channels_to_join = self._filter_add_members(target_partners)
+
+        # Existing channel partners
+        existing_channel_partners = self.env['slide.channel.partner'].with_context(active_test=False).sudo().search([
+            ('channel_id', 'in', channels_to_join.ids),
```

Comment:
If there are no channels_to_join, we could skip the search (was the purpose of the old "if").